design/key_derivation.md

```diff
@@ -0,0 +1,42 @@
+# Key Derivation design
+
+## Scope
+
+Obscuro blockchain is seeded with a network Key.
+This network key is shared between enclaves upon successful attestation.
+
+Transactions are encrypted in the rollups that make the Obscuro blockchain.
+Each rollup uses a key to encrypt the Transactions.
+This key is derived from the network Key.
+
+## Requirements
+
+* Each rollup is encrypted with a unique key
+  * Each unique key is derived from the network key
+  * Each unique key is not able to disclose other unique keys
+  * Each unique key is not able to disclose the network key
+  * Each unique key is deterministic give the rollup height
+* Each enclave is able to determine the unique key given the enclave secret and rollup height
+
+## Design
+
+Use HKDF standardized in https://www.rfc-editor.org/rfc/rfc5869.
```

Comment:
I'd suggest expanding this section a bit. 
First a bit of context.
Followed by the options.
Each with pros, cons.

Reply:
Can do 👍